### ISSUE TITLE
Add metric roc_auc_weighted

### ISSUE THREAD

**Issue by sonichi** [microsoft/FLAML#774]
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html
Set average="weighted".
Find all occurrences of "roc_auc" in ml.py, add "roc_auc_weighted" accordingly, and update the documentation website and test.
